File: AndroidProduct/app/src/main/java/com/kai/kaiproductanndroid/APIController.java
package com.kai.kaiproductanndroid;
import com.Tools.API.BaseAPICallBack;
import com.Tools.API.FetchJsonTask;
import com.Tools.API.PostJsonTask;
import com.Tools.mLog;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.Iterator;

public class APIController
{
    public void GetKaiTest(JSONObject input, BaseAPICallBack callBack) {
        sendAPI("kai_test", input, callBack);
    }

    public void GetFgoGetRole(JSONObject input, BaseAPICallBack callBack) {
        sendAPI("role", input, callBack);
    }

    public void GetFgoGetGender(JSONObject input, BaseAPICallBack callBack) {
        sendAPI("gender", input, callBack);
    }

    public void GetFgoGetProfession(JSONObject input, BaseAPICallBack callBack) {
        sendAPI("profession", input, callBack);
    }

    public void GetFgoRoleLevelAttr(JSONObject input, BaseAPICallBack callBack)
    {
        sendAPI("level_attr",input,callBack);
    }

    public void GetFgoProfessionAtkRate(JSONObject input,BaseAPICallBack callBack)
    {
        sendAPI("profession_atk_rate",input,callBack);
    }

    public void  GetFgoMapPos(JSONObject input,BaseAPICallBack callBack)
    {
        sendAPI("map_pos",input,callBack);
    }

    public void GetAiTalkGroup(JSONObject input,BaseAPICallBack callBack)
    {
        sendAPI("talk_group",input,callBack);
    }

    public void GetAiTalkContent(JSONObject input,BaseAPICallBack callBack)
    {
        sendAPI("talk_content",input,callBack);
    }

    public void AskAiTalk(JSONObject input,BaseAPICallBack callBack)
    {
        postAPI("talk_group_ask",input,callBack);
    }

    private void sendAPI(String url,JSONObject input,BaseAPICallBack callBack) {
        String p = "";
        if(input.length() > 0) {
            for (Iterator<String> it = input.keys(); it.hasNext(); ) {
                String key = it.next();
                String value = null;
                try {
                    value = input.getString(key);
                } catch (JSONException e) {
                    throw new RuntimeException(e);
                }
                if(!p.isEmpty())
                {
                    p += "&";
                }
                else
                {
                    p += "?";
                }
                p += key + "=" +value;
            }
        }
        new FetchJsonTask(config.API_DOMAIN + url + p, callBack).execute();
    }

    private void postAPI(String url,JSONObject input,BaseAPICallBack callBack)
    {
        new PostJsonTask(config.API_DOMAIN + url,input, callBack).execute();
    }
}

File: fristUiKit/fristUiKit/API/APIController.swift
//
//  APIController.swift
//  fristSwiftUI
//
//  Created by 尤凱 on 2025/4/10.
//

import Foundation

class APIController
{
    typealias APICallbackClosure = (APIResult) -> Void // 定義一個閉包型別
    
    func GetKaiTest(input: [String: Any], callBack: @escaping APICallbackClosure)
    {
        sendApi(url:"kai_test",input: input,callBack: callBack);
    }

    func GetRole(input: [String: Any], callBack: @escaping APICallbackClosure)
    {
        sendApi(url:"role",input: input,callBack: callBack);
    }
    
    func GetRoleResource(input:[String:Any],callBack: @escaping APICallbackClosure)
    {
        sendApi(url: "role_resource", input: input, callBack: callBack);
    }
    
    func GetGender(input: [String: Any], callBack: @escaping APICallbackClosure)
    {
        sendApi(url:"gender",input: input,callBack: callBack);
    }
    
    func GetProfession(input: [String: Any], callBack: @escaping APICallbackClosure)
    {
        sendApi(url:"profession",input: input,callBack: callBack);
    }
    
    func GetFgoRoleLevelAttr(input : [String:Any],callBack: @escaping APICallbackClosure)
    {
        sendApi(url:"level_attr" , input:input, callBack: callBack);
    }
    
    func GetFgoProfessionAtkRate(input : [String:Any],callBack: @escaping APICallbackClosure)
    {
        sendApi(url: "profession_atk_rate", input: input, callBack: callBack);
    }

    func GetFgoMapPos(input : [String:Any],callBack: @escaping APICallbackClosure)
    {
        sendApi(url: "map_pos", input: input, callBack: callBack);
    }

    func GetAiTalkGroup(input : [String:Any],callBack: @escaping APICallbackClosure)
    {
        sendApi(url: "talk_group", input: input, callBack: callBack);
    }

    func GetAiTalkContent(input : [String:Any],callBack: @escaping APICallbackClosure)
    {
        sendApi(url: "talk_content", input: input, callBack: callBack);
    }
    
    func AskAiTalk(input : [String:Any],callBack: @escaping APICallbackClosure)
    {
        postApi(url: "talk_group_ask", input: input, callBack: callBack);
    }
    
    func sendApi(url:String,input: [String: Any], callBack : @escaping APICallbackClosure)
    {
        print(input);
        
        var p : String = "";
        if(input.count > 0)
        {
            for key in input.keys
            {
                if(!p.isEmpty)
                {
                    p += "&";
                }
                else
                {
                    p += "?";
                }
                if let value = input[key] as? Int
                {
                    p += "\(key)=\(value)";
                }
                if let value = input[key] as? String
                {
                    p += "\(key)=\(value)";
                }
                if let value = input[key] as? Float
                {
                    p += "\(key)=\(value)";
                }
                if let value = input[key] as? Double
                {
                    p += "\(key)=\(value)";
                }
            }
        }
        
        
        let uri = config.API_DOMAIN + url + p;
        
        print(uri);
        
        fetchJSON(from: uri) { result in
            switch result {
            case .success(let json):
                var apiresult = SwiftAPIResult(data: json);
                print("\(apiresult)");
                callBack(apiresult);
            case .failure(let error):
                var errorresult = SwiftAPIResult();
                callBack(errorresult);
            }}
    }
    
    func postApi(url:String,input: [String: Any], callBack : @escaping APICallbackClosure)
    {
        print(input);
        
        let uri = config.API_DOMAIN + url;
        
        print(uri);
        
        postJSON(to: uri,body: input) { result in
            switch result {
            case .success(let json):
                var apiresult = SwiftAPIResult(data: json);
                print("\(apiresult)");
                callBack(apiresult);
            case .failure(let error):
                var errorresult = SwiftAPIResult();
                callBack(errorresult);
            }}
    }}
